code/game/objects/structures/simple_doors.dm

```diff
@@ -82,6 +82,11 @@
 /obj/structure/simple_door/attack_hand(mob/user as mob)
 	return TryToSwitchState(user)
 
+/obj/structure/simple_door/attack_generic(mob/user)
+	if(istype(user, /mob/living/simple_animal/construct)) // don't know of any other attack_generic smart enough to open doors
```

Comment:
There's some robotic simple mobs like hivebots, should they be able to open these?

Maybe it should be a var.. or flag... or something. Idk where you'd put it though exactly.

Reply:
nahh, i don't think so. they're not really player controlled either. if someone does add to them later on, they can take a peek.